library/src/main/java/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadFragment.kt

```diff
@@ -68,17 +73,16 @@ internal class TransactionPayloadFragment :
             viewLifecycleOwner,
             Observer { transaction ->
                 if (transaction == null) return@Observer
-                PayloadLoaderTask(this).execute(Pair(type, transaction))
+                CoroutineScope(Dispatchers.Main).launch {
+                    showProgress()
+                    val result = processPayload(type, transaction)
+                    responseRecyclerView.adapter = TransactionBodyAdapter(result)
+                    hideProgress()
+                }
             }
         )
     }
 
-    override fun onDestroyView() {
-        super.onDestroyView()
-        payloadLoaderTask?.cancel(true)
-        fileSaverTask?.cancel(true)
-    }
-
```

Comment:
Jobs that replace now taks still should be tied the lifecycle (or to a view model).

Reply:
Yeah. I'll add supervisor job.